### DISCUSSION TITLE
neovim configuration: lazy loading for cmp-nvim-lsp

### DISCUSSION THREAD

**Discussion by uraza** [numToStr/dotfiles#15] [Ideas]
Hello.

I came across your neovim configuration on Reddit and found it really useful to learn about Packer's lazy loading capabilities.
This helped me speed up my Neovim startup time quite a bit, thank you for that! :)

When having a look at your configuration, I noticed this section where one plugin is not loaded lazily:

```
use({
    'neovim/nvim-lspconfig',
    event = 'BufRead',
    config = function()
        require('numToStr.plugins.lsp.servers')
    end,
    requires = {
        {
            -- WARN: Unfortunately we won't be able to lazy load this
            'hrsh7th/cmp-nvim-lsp',
        },
    },
})
```

On my side, it looks like changing it into the following piece of code works as expected:

```
use({
    {
        'hrsh7th/cmp-nvim-lsp',
        event = 'BufReadPre'
    },
    {
        'neovim/nvim-lspconfig',
        after = 'cmp-nvim-lsp',
        config = function()
            require('numToStr.plugins.lsp.servers')
        end
    }
})
```

Is that working on your side as well?
Please note that I am not using your configuration as is, so I may have missed something.

Thanks.

**Comment by numToStr** [comment]
I am glad that my config helped you in setting up your own.

IIRC this https://github.com/hrsh7th/nvim-cmp/issues/65#issuecomment-905692825 was the reason for not lazy-loading it, this was more than a year ago and I didn't change my LSP setup since then 😅. Although I am sure things have changed, as you also pointed out. Anyways I'll try your solution and see what happens.

✌️

**Comment by EpsilonKu** [comment]
I recommend to look on NVchad lazy load. Also with lazy.nvim less problems.
